src/pages/dashboard/feedstock/[id].js

```diff
@@ -40,7 +46,12 @@ const Feedstock = () => {
 
   if (fsError || metaError || datasetsError)
     return (
-      <Layout>
+      <Layout
+        title={meta?.title}
+        description={meta?.description}
+        image={`${url}/api/og/feedstock?id=${id}&spec=${spec}`}
+        url={`${url}/feedstock/${id}`}
```

Comment:
@katamartin, i'm curious... 🧐 

I am trying to populate the meta tags used by opengraph using the `id` and `spec`. The `id `and `spec`, however, are undefined here since I'm generating meta tags server-side. Would it even be possible to pre-render these server-side?

Reply:
Oooh good question. Hm so looking at this page, it seems like the meta tags _are_ being generated client-side, but they are initially rendered without `id` or `spec`. First, because the page has not yet been hydrated with the routing information. Then, because the `spec` has not been fetched.

Of course, the `id` and `spec` _do eventually_ get populated and show up in the `meta` tags, but I think might not get picked up properly in social previews because they aren't included in the initial values. I think the solution here might involve delaying rendering the `Layout` (which renders the `meta` tags) until `id` and `spec` become available. This doesn't seem ideal for the performance of the actual page, but I suspect that could be improved (e.g. by only requiring `id` to be hydrated and also fetching `spec` in the card generator) after confirming that this fix directionally works.

Reply:
See https://github.com/pangeo-forge/pangeo-forge.org/pull/225#discussion_r998562559